cmd/main.go

```diff
@@ -286,16 +289,45 @@ func ProcessOutChannel(wg *sync.WaitGroup, scConfig *common.SCConfiguration) {
 									log.Infof("post events %s to subscriber %s", d.Address, endPointURI)
 									// make sure event ID is unique
 									event.ID = uuid.New().String()
-									err = restClient.PostCloudEvent(endPointURI, *d.Data)
-									postHandler(err, sub.EndPointURI, d.Address)
+									var status, numSubDeleted int
+									status, err = restClient.PostCloudEvent(endPointURI, *d.Data)
+									if err != nil {
+										log.Errorf("error posting event at %s : %s", endPointURI, err)
+										localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.FAILED)
+
+										// POST return DNSError if server is not reachable
+										var dnsError *net.DNSError
+										// Capture DNS error "lookup consumer-events-subscription-service.cloud-events.svc.cluster.local: no such host"
+										// or timeout error "context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
+										if errors.As(err, &dnsError) || os.IsTimeout(err) {
+											// has subscriber failed to connect for 10 times delete the subscribers
+											if scConfig.SubscriberAPI.IncFailCountToFail(clientID) {
+												log.Errorf("connection lost for address %s, proceed to clean up subscription", d.Address)
+												if numSubDeleted, err = scConfig.SubscriberAPI.DeleteAllSubscriptionsForClient(clientID); err != nil {
+													log.Errorf("failed to delete all subscriptions for client %s: %s", clientID.String(), err.Error())
+												} else {
+													cleanupConfigMap(d.ClientID)
```

Comment:
does this clean up inmemory database of the publisher from the store map ?

Reply:
added cleanup

Reply:
in-memory db is only cleaned up when deleting all subs for all clients. In this case for client timeout, only deleting from SubscribnerAPI will be enough. The events will be stopped for this client.